pyartcd/pyartcd/pipelines/ocp4.py

```diff
@@ -777,10 +777,17 @@ async def _sync_images(self):
 
         records = record_log.get('build', [])
         operator_nvrs = []
-
+        operator_nvrs_map = {}
+        operator_deps = []
         for record in records:
             if record['has_olm_bundle'] == '1' and record['status'] == '0' and record.get('nvrs', None):
                 operator_nvrs.append(record['nvrs'].split(',')[0])
+            if record['status'] != '0' and record.get('nvrs', None) and record.get('dependents', None):
+                operator_deps.extend([dep for dep in record['dependents'].split(',') if dep not in operator_deps])
+            operator_nvrs_map[record['distgit']] = record.get('nvrs')
+        for dep in operator_deps:
+            if dep in operator_nvrs_map and operator_nvrs_map[dep] in operator_nvrs:
```

Comment:
This code is very hard to read. I understand the intent but would suggest to make it clear as to what we're doing. 

I would suggest to first go through records to collect all `failed_dependent_images` , and then have a second loop through records to collect `operator_nvrs` based on the above. This way we do not have to `.remove()`